payment_ogone_compassion/models/payment.py

```diff
@@ -0,0 +1,140 @@
+                'lastname': values['LASTNAME'],
+                'email': values['EMAIL'],
+                'phone': values['OWNERTELNO'],
+                'street': values['OWNERADDRESS'],
+                'city': values['OWNERTOWN'],
+                'zip': values['OWNERZIP'],
+                'country_id': country_id
+            })
+
+        invoice_vals = {'partner_id': partner.id,
+                        'currency_id': currency_id
+                        }
+
+        invoice = self.env['account.invoice'].create(invoice_vals)
+        invoice_line_vals = {'quantity': 1.0,
+                             'price_unit': values['AMOUNT'],
+                             'invoice_id': invoice.id,
+                             'account_id': 2775,  # This is specific to
+                             # muskathlon. Should be change for something
+                             # dependant on some input
+                             'name': 'Payment by ogone',
+                             'product_id': self.env['product.product'].search(
+                                 [('name', '=', values['product_name'])]).id,
+                             }
+        invoice_line = self.env['account.invoice.line'].create(
+            invoice_line_vals)
+
+        if values['ambassador'] and values['event_id']:
+            # We are dealing with a payment for a muskathlon -> updating
+            # the invoice line to add values so that the gift will be
+            # registered toward the muskathlon goal
+            self.update_invoice_line_for_muksathlon(invoice_line, values[
+                'ambassador'], values['event_id'])
+
+        tx_data = {
+            'state': u'draft',
+            'acquirer_id': ogone.id,
+            'partner_id': partner.id,
+            'currency_id': currency_id,
+            'partner_country_id': country_id,
+            'partner_name': partner.firstname + partner.lastname,
+            'partner_phone': values['OWNERTELNO'],
+            'partner_addres': values['OWNERADDRESS'],
+            'partner_lang': values['LANGUAGE'],
+            'partner_email': values['EMAIL'],
+            'partner_zip': values['OWNERZIP'],
+            'partner_city': values['OWNERTOWN'],
+            'partner_state': None,
+            'amount': float(values['AMOUNT']),
+            'reference': unicode(invoice_line.id)
+        }
+
+        tx = self.env['payment.transaction'].create(tx_data)
+        tx_data.update({
+            'partner_country': tx.partner_country_id,
+            'currency': tx.currency_id
+        })
+
+        res = ogone.ogone_form_generate_values(tx_data)
+
+        # filter field to send to Ogone
+        res = {k: v for k, v in res.iteritems() if k.isupper()}
+        del res['PARAMPLUS']
+        return ogone.ogone_get_form_action_url(), res
+
+    def validate_invoice_line(self, invoice_id):
+        invoice_line = self.env['account.invoice.line'].search([('id', '=',
```

Comment:
Here the browse method does what you do with search. You can directly put the id and it fetches the record.

Reply:
Done